arangod/Aql/AstNode.cpp

```diff
@@ -2620,6 +2622,18 @@ void AstNode::removeMembersInOtherAndNode(AstNode const* other) {
   }
 }
 
+void AstNode::markFinalized(AstNode const* subtreeRoot) {
```

Comment:
It is really-counter intuitive that `markFinalized` is getting a const AstNode as its input.
Can we pass a non-const input to it?

Reply:
Will fix.